zipline/pipeline/loaders/blaze/_core.pyx

```diff
@@ -0,0 +1,651 @@
+        _ffill_missing_value_2d_inplace_impl[np.int64_t](
+            array,
+            missing_value,
+            is_missing_value[np.int64_t],
+        )
+    elif kind == 'u':
+        _ffill_missing_value_2d_inplace_impl[np.uint8_t](
+            array,
+            missing_value,
+            is_missing_value[np.uint8_t],
+        )
+    elif kind == 'M':
+        _ffill_missing_value_2d_inplace_impl[np.int64_t](
+            array.view('int64'),
+            missing_value.astype('int64'),
+            is_missing_value[np.int64_t],
+        )
+    elif kind == 'f':
+        if isnan(missing_value):
+            _ffill_missing_value_2d_inplace_impl[np.float64_t](
+                array,
+                missing_value,
+                is_missing_nan,
+            )
+        else:
+            _ffill_missing_value_2d_inplace_impl[np.float64_t](
+                array,
+                missing_value,
+                is_missing_value[np.float64_t],
+            )
+    elif kind == 'O':
+        _ffill_missing_value_2d_inplace_impl[object](
+            array,
+            missing_value,
+            is_missing_value[object],
+        )
+    else:
+        raise TypeError('unknown column dtype: %r' % array.dtype)
+
+
+
+@cython.final
+cdef class AsAdjustedArray:
+    """Marker type for array_for_column which enables the AdjustedArray
+    path.
+    """
+
+
+@cython.final
+cdef class AsBaselineArray:
+    """Marker type for array_for_column which enables the baseline array
+    path.
+    """
+
+
+ctypedef fused AsArrayKind:
+    AsAdjustedArray
+    AsBaselineArray
+
+
+cdef _array_for_column_impl(object dtype,
+                            np.ndarray[column_type, ndim=2] out_array,
+                            Py_ssize_t size,
+                            np.ndarray[np.int64_t] ts_ixs,
+                            np.ndarray[np.int64_t] asof_ixs,
+                            np.ndarray[np.int64_t] sids,
+                            dict column_ixs,
+                            object mask,
+                            np.ndarray[column_type] input_array,
+                            column_type missing_value,
+                            bint is_missing(column_type, column_type),
+                            AsArrayKind _):
+    cdef column_type value
+    cdef np.int64_t ts_ix
+    cdef np.int64_t asof_ix
+    cdef np.int64_t sid
+    cdef np.int64_t column_ix
+
+    cdef PyObject* adjustments_list_ptr
+    cdef list adjustments_list
+
+    cdef list non_null_ixs
+    cdef dict non_null_ixs_by_sid = {sid: [] for sid in sids}
+    cdef dict adjustments
+
+    cdef Py_ssize_t out_of_bounds_ix = len(out_array)
+
+    if AsArrayKind is AsAdjustedArray:
+        adjustments = {}
+
+    cdef set categories
+    if column_type is object:
+        # for object columns we need to maintain the unique values for the
+        # categories
+        categories = set()
+
+    cdef Py_ssize_t n
+    for n in range(size if len(out_array) else 0):
```

Comment:
this can happen when attempting to build checkpoints for a dataset that doesn't span more than a month. It should just return an empty array